### PR TITLE
[WFLY-14980] add default 60s r/o timeout to listeners

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WFLY-14980

Follows on: https://github.com/wildfly/wildfly/pull/15041
Supersedes: https://github.com/wildfly/wildfly/pull/14595

### CODE REVIEW

**Comment by ropalka** [comment]
@darranl please review

**Comment by bstansberry** [comment]
@ropalka The needs a management API version bump and a transformer, so legacy host controller runs with a 60 s timeout if  no value is explicitly configured.

**Comment by ropalka** [comment]
Hello @bstansberry , the bump is already in, see: https://github.com/wildfly/wildfly/commit/a1f9ad78f6c960e824a2ca007a40593706c2d060
What about transformers, we decided to drop them, didn't we https://github.com/wildfly/wildfly/commit/ebef9a93f71ad8ea52e286d9983dce8833592d9e ?

**Comment by bstansberry** [comment]
@ropalka No, we didn't drop transformers, we just reduced the number of old releases we transform to to WF 23 or later. So this needs transformers.

**Comment by ropalka** [comment]
Fixed @bstansberry and thanks for the clarification.

**Comment by ropalka** [comment]
This should get it into WildFly @bstansberry before we will release it. ( @fl4via CC )

**Comment by ropalka** [comment]
Just an FYI this is a long pending PR @bstansberry .

**Comment by bstansberry** [comment]
Thanks @ropalka

File: undertow/src/main/java/org/wildfly/extension/undertow/UndertowTransformers.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    private static void registerTransformersWildFly23(ResourceTransformationDescriptionBuilder subsystemBuilder) {
+        final ResourceTransformationDescriptionBuilder serverBuilder = subsystemBuilder.addChildResource(UndertowExtension.SERVER_PATH);
+
+        final AttributeTransformationDescriptionBuilder http = serverBuilder.addChildResource(UndertowExtension.HTTP_LISTENER_PATH).getAttributeBuilder()
+                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE, WRITE_TIMEOUT)
+                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE, READ_TIMEOUT)
+                .addRejectCheck(new SimpleRejectAttributeChecker(ModelNode.TRUE), WRITE_TIMEOUT.getName())
+                .addRejectCheck(new SimpleRejectAttributeChecker(ModelNode.TRUE), READ_TIMEOUT.getName());
```
Comment (pferraro) [inline undertow/src/main/java/org/wildfly/extension/undertow/UndertowTransformers.java:67]: This transformation does not look correct to me.  A new default value should not result in any rejections, but rather use an AttributeConverter.
I've fixed this (and included missing tests) in this PR: https://github.com/wildfly/wildfly/pull/16455

File: undertow/src/main/resources/schema/wildfly-undertow_13_0.xsd [hunk 2]
```diff
@@ -74,8 +74,8 @@
         <xs:attribute name="send-buffer" type="xs:int"/>
         <xs:attribute name="tcp-backlog" type="xs:int" default="10000"/>
         <xs:attribute name="tcp-keep-alive" type="xs:boolean"/>
-        <xs:attribute name="read-timeout" type="xs:long"/>
-        <xs:attribute name="write-timeout" type="xs:long"/>
+        <xs:attribute name="read-timeout" type="xs:long" default="60000"/>
+        <xs:attribute name="write-timeout" type="xs:long" default="60000"/>
```
Comment (pferraro) [inline undertow/src/main/resources/schema/wildfly-undertow_13_0.xsd:78]: Version 13 was introduced in WF27.  Modifications to XSDs are generally done in a new version.
I've patched this here: https://github.com/wildfly/wildfly/pull/16455

**Pull request merged**